tool_operate: Don't append a filename to an SMTP URL

jay opened the pull request:
Prior to this change sending a message via SMTP file upload could cause the filename to be sent as the EHLO, for example:

```
curl -v -u user:pass --mail-rcpt my_username_at_gmail.com -T mail.txt \
     smtps://smtp.gmail.com
```
curl would change the URL to `smtps://smtp.gmail.com/mail.txt` and since SMTP URLs with a path have that sent as the EHLO that would then cause:
```
EHLO mail.txt
```

bagder commented:
Not sure this is worth it. Did anyone ever complain? And for a good smtp URL a user should provide the name as path anyway since curl cannot figure out accurately by itself.

jay commented:
> And for a good smtp URL a user should provide the name as path anyway since curl cannot figure out accurately by itself.

I'd guess many users would just smtp://smtp.foo.com and not worry about providing a path, but I don't have anything to back that up. Regardless my position is it's wrong to append the filename to an SMTP URL since that ultimately means it will be used it in the EHLO.

This is out of draft now, I've moved the detection code to a helper function and rebased on master.
